cdflow_commands/ecs_monitor.py

```diff
@@ -0,0 +1,157 @@
+from hashlib import sha1
+from functools import lru_cache
+from time import sleep, time
+
+from cdflow_commands.logger import logger
+
+
+def build_service_name(environment, component):
+    service_name = '{}-{}'.format(environment, component)
+    if len(service_name) > 32:
+        service_name_hash = sha1(service_name.encode('utf-8')).hexdigest()
+        service_name = '{}tf{}'.format(
+            service_name[:24], service_name_hash[:4]
+        )
+    return service_name
+
+
+class ECSMonitor():
+
+    _TIMEOUT = 600
+    _ITERATION_DELAY = 15
+
+    def __init__(self, ecs_event_iterator):
+        self._ecs_event_iterator = ecs_event_iterator
+
+    def wait(self):
+        start = time()
+        for event in self._ecs_event_iterator:
+            if time() - start > self._TIMEOUT:
+                logger.error(
+                    'Deployment timed out - '
+                    'didn\'t complete within {} seconds'.format(self._TIMEOUT)
+                )
+                raise TimeoutError
+            if event.done:
+                logger.info('Deployment complete - running: {}'.format(
+                    event.running
+                ))
+                return True
+            logger.info(
```

Comment:
As discussed, move above so you always get one of these, even if it runs really quickly.